version: '3.8'

services:
  postgres:
    image: postgres:16.4
    command: [postgres, --max_connections=100]
    restart: always
    environment:
      - POSTGRES_USER=socger
      - POSTGRES_PASSWORD=sasa
      - POSTGRES_DB=n8n_bd
      - POSTGRES_NON_ROOT_USER=socger_nonroot
      - POSTGRES_NON_ROOT_PASSWORD=sasa_nonroot
    volumes:
      - postgresql_data:/var/lib/postgresql/data
      # el init-data.sh siguiente se cambió por la siguiente línea de después
      #- ./init-data.sh:/docker-entrypoint-initdb.d/init-data.sh
      - /docker/sh_creados/init-data.sh:/docker-entrypoint-initdb.d/init-data.sh
    ports:
      - "5432:5432"  # <-- Aquí exponemos el puerto PostgreSQL
    networks:
      - shared_backend
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U socger -d n8n_bd"]
      interval: 10s
      timeout: 5s
      retries: 5
      start_period: 10s
    #deploy:
      #placement:
        #constraints:
          #- node.role == manager
    deploy:
      mode: replicated
      replicas: 1
      placement:
        constraints:
          - node.role == manager
      resources:
        limits:
          cpus: "1"
          memory: 2048M
      restart_policy:
        condition: on-failure
        delay: 5s
        max_attempts: 3
        window: 120s

networks:
  shared_backend:
    external: true

volumes:
  postgresql_data:
    driver: local
    driver_opts:
      type: none
      device: /docker/postgresql/data
      o: bind
